### PR TITLE
Fixing `Asset`'s `<=>` method

### PR DESCRIPTION
**Pull request by myk-kyryk**
# Related issue
https://github.com/Shopify/shopify/issues/233260

# Description
Fixing `Asset`'s `<=>` method to work correctly when used for comparing (`==`) with instances of other classes.

Adding tests to cover different edge cases.
Previous implementation would fail if comparing Asset object with any other object which doesn't respond to `cloud` and `key` methods.
Additionally we used logical `AND` to combine check for `cloud` and `key`. This would work fine if the comparison would be done via `==`,
but with `<=>` we are getting number back (-1, 0, or 1), so combining these via logical `AND` was producing unexpected results.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ Gemfile.lock
 .rvmrc
 *.gem
 tags
+.idea
```
Comment (rafaelfranca) [inline .gitignore]:
```suggestion
```

Comment (myk-kyryk) [reply .gitignore]: Hi @rafaelfranca 
Can I ask you why you don't like this change?

Comment (rafaelfranca) [reply .gitignore]: Because this file is not part of the project, so it should be configured in your global ignore file, not in the project specific git ignore.

Comment (myk-kyryk) [reply .gitignore]: Got it 👍

File: lib/asset_cloud/asset.rb [hunk 2]
```diff
@@ -34,7 +34,13 @@ def self.at(cloud, key, value = nil, metadata = nil, &block)
     end
 
     def <=>(other)
-      cloud.object_id <=> other.cloud.object_id && key <=> other.key
+      return 1 unless other.is_a?(Asset)
+      compare_cloud = cloud.object_id <=> other.cloud.object_id
```
Comment (Edouard-chin) [inline lib/asset_cloud/asset.rb]: Not sure if I understand why we use the `<=>` operator here? Shouldn't it just be `cloud.object_id == other.cloud.object_id && key == other.key`

Ruby `==` operator will call `<=>` if needed but I don't think we need to do it ourselves

Comment (rafaelfranca) [reply lib/asset_cloud/asset.rb]: The `<=>` can't return `true/false` it needs to return 0, 1 and -1. If we use `==` we will return `true/false`.

Comment (Edouard-chin) [reply lib/asset_cloud/asset.rb]: Ah yeah that's true
